docs/devguide/dtypes.rst

```diff
@@ -1,2 +1,77 @@
 Implementing a New Data Type
 ============================
+
+COMPAS data types are classes that are based on :class:`compas.data.Data`.
+
+Data types can be serialised to JSON with
+
+* :func:`compas.json_dump`
+* :func:`compas.json_dumps`
+* :func:`compas.json_dumpz`
+
+and deserialised with the corresponding "load" functions
+
+* :func:`compas.json_load`
+* :func:`compas.json_loads`
+* :func:`compas.json_loadz`
+
+All geometry objects and data structures,
+and also, for example, the visualisation scene,
+are data types.
+
+
+Creating a new data type
+========================
+
+A new data type can be created in a few simple steps.
+
+.. code-block:: python
+
+    class CustomData(Data):
+
+        def __init__(self, a=None, b=None, name=None)
+            super().__init__(name=name)
+            self.a = a
+            self.b = b
+
+        @property
+        def __data__(self):
+            data = super().__data__
```

Comment:
This syntax only works for Python 3, right? In particular the property getting I think